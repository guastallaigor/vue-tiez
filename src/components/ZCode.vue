<template>
  <div class="z-code" :class="getClasses" :style="`width: ${width};`">
    <template v-if="content">{{ content }}</template>
    <slot v-else />
  </div>
</template>

<script>
export default {
  name: 'z-code',
  props: {
    content: {
      type: String,
      default: ''
    },
    withDolar: {
      type: Boolean,
      default: false
    },
    width: {
      type: String,
      default: 'auto'
    },
    dark: {
      type: Boolean,
      default: false
    }
  },
  computed: {
    getClasses () {
      const { withDolar, dark } = this

      return {
        'with-dolar': withDolar,
        dark
      }
    }
  }
}
</script>
