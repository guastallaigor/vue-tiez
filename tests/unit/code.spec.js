import { shallowMount } from '@vue/test-utils'
import ZCode from '../../src/components/ZCode'

describe('When I create the ZCode component', () => {
  const createComponent = (propsData = {}, slot) => {
    return shallowMount(ZCode, {
      propsData,
      slots: {
        default: slot || 'ZCode'
      }
    })
  }

  it('should be a Vue instance', () => {
    const wrapper = createComponent({ content: 'Content' })
    expect(wrapper.find(ZCode).isVueInstance()).toBe(true)
  })

  it('should have a content', () => {
    const wrapper = createComponent({ content: 'Content' })
    const code = wrapper.find('.z-code')
    expect(code.exists()).toBe(true)
    expect(code.classes().length).toBe(1)
    expect(code.classes()).toEqual(['z-code'])
    expect(code.html()).toBe('<div class="z-code">Content</div>')
  })

  it('should have a content with dolar', () => {
    const wrapper = createComponent({ content: 'Content', withDolar: true })
    const code = wrapper.find('.z-code')
    expect(code.exists()).toBe(true)
    expect(code.classes().length).toBe(2)
    expect(code.classes()).toEqual(['z-code', 'with-dolar'])
    expect(code.html()).toBe('<div class="z-code with-dolar">Content</div>')
  })

  it('should have a content and dark class', () => {
    const wrapper = createComponent({ content: 'Content', dark: true })
    const code = wrapper.find('.z-code')
    expect(code.exists()).toBe(true)
    expect(code.classes().length).toBe(2)
    expect(code.classes()).toEqual(['z-code', 'dark'])
    expect(code.html()).toBe('<div class="z-code dark">Content</div>')
  })

  it('should have a slot instead of a content', () => {
    const wrapper = createComponent({}, '<span>test with slot</span>')
    const code = wrapper.find('.z-code')
    expect(code.exists()).toBe(true)
    expect(code.classes().length).toBe(1)
    expect(code.classes()).toEqual(['z-code'])
    expect(code.html()).toBe('<div class="z-code"><span>test with slot</span></div>')
  })

  it('should match snapshot', () => {
    const wrapper = createComponent({
      content: 'Content',
      withDolar: true,
      dark: false
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
