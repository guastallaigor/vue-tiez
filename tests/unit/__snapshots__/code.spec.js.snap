// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When I create the ZCode component should match snapshot 1`] = `<div class="z-code with-dolar" style="width: auto;">Content</div>`;
